mateweather/mateweather-dialog.c

```diff
@@ -502,6 +502,35 @@ static PangoFontDescription* get_system_monospace_font(void)
     return desc;
 }
 
+static void
+override_widget_font (GtkWidget   *widget,
+                      const gchar *font)
+{
+    gchar          *css;
+    GtkCssProvider *provider;
+    gchar          *tempsize;
+
+    provider = gtk_css_provider_new ();
+    tempsize = g_strdup (font);
+
+    g_strreverse (tempsize);
+    g_strcanon (tempsize, "1234567890", '\0');
+    g_strreverse (tempsize);
+
+    gchar tempfont [strlen (font)];
+    strcpy (tempfont, font);
+    tempfont [strlen (font) - strlen (tempsize)] = 0;
+
+    css = g_strdup_printf ("* { font-family: %s; font-size: %spt; }", tempfont, tempsize);
+    gtk_css_provider_load_from_data (provider, css, -1, NULL);
+    g_free (css);
+
+    gtk_style_context_add_provider (gtk_widget_get_style_context (widget),
```

Comment:
`mateweather_dialog_update` sounds like something that can be called multiple times. Why every time create new provider?